File: Darwin/ShowcaseUITests/NavigationStackPlaygroundUITests.swift
// Copyright 2023–2025 Skip
import XCTest

@MainActor
final class NavigationStackPlaygroundUITests: XCTestCase, Sendable {
    private var app: XCUIApplication!

    override func setUp() async throws {
        try await super.setUp()
        continueAfterFailure = false
        app = XCUIApplication()
        app.launchArguments.append("-UITestingResetShowcaseSearch")
        app.launch()
    }

    func testNavigationStackPlaygroundShowsPopButton() throws {
        app.tabBars.buttons["Showcase"].tap()

        let searchField = app.searchFields.firstMatch
        XCTAssertTrue(searchField.waitForExistence(timeout: 5), "Search field should appear on Showcase tab")
        searchField.tap()
        searchField.typeText("NavigationStack")

        let row = app.staticTexts["NavigationStack"]
        XCTAssertTrue(row.waitForExistence(timeout: 5), "Filtered list should include NavigationStack")
        row.tap()

        XCTAssertTrue(app.buttons["Pop"].waitForExistence(timeout: 5), "NavigationStack playground should show Pop button")
    }
}

File: Android/app/src/androidTest/kotlin/showcase/module/NavigationStackPlaygroundUITests.kt
package showcase.module

import skip.lib.*
import skip.unit.*
import skip.uiautomation.*

@org.junit.runner.RunWith(androidx.test.ext.junit.runners.AndroidJUnit4::class)
class NavigationStackPlaygroundUITests : XCTestCase {
    @Test
    fun testNavigationStackPlaygroundShowsPopButton() {
        val app = XCUIApplication()
        app.launch()

        app.tabBars.buttons["Showcase"].tap()

        val searchField = app.searchFields.firstMatch
        XCTAssertTrue(searchField.waitForExistence(timeout = 5.0), "Search field should appear on Showcase tab")
        searchField.tap()
        searchField.typeText("NavigationStack")

        val row = app.staticTexts["NavigationStack"]
        XCTAssertTrue(row.waitForExistence(timeout = 5.0), "Filtered list should include NavigationStack")
        row.tap()

        XCTAssertTrue(app.buttons["Pop"].waitForExistence(timeout = 5.0), "NavigationStack playground should show Pop button")
    }
}
